### PR TITLE
Remove unused extension method

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Features/Diagnostics/IDiagnosticServiceExtensions.cs [hunk 1]
```diff
@@ -5,16 +5,11 @@
...
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.PooledObjects;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
     internal static class IDiagnosticServiceExtensions
     {
-        public static ValueTask<ImmutableArray<DiagnosticData>> GetPullDiagnosticsAsync(this IDiagnosticService service, DiagnosticBucket bucket, bool includeSuppressedDiagnostics, DiagnosticMode diagnosticMode, CancellationToken cancellationToken)
-            => service.GetPullDiagnosticsAsync(bucket.Workspace, bucket.ProjectId, bucket.DocumentId, bucket.Id, includeSuppressedDiagnostics, diagnosticMode, cancellationToken);
```
Comment (Youssef1313) [inline src/Features/LanguageServer/Protocol/Features/Diagnostics/IDiagnosticServiceExtensions.cs:16]: Is `GetPullDiagnosticsAsync` called anywhere else?

Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Features/Diagnostics/IDiagnosticServiceExtensions.cs:16]: Yes :)

**Comment by Youssef1313** [comment]
> Yes :)

I just noticed the other callsite and have deleted the comment before your comment appeared. :)

**Review by Youssef1313 — APPROVED** [review]
🎉

**Review by akhera99 — APPROVED** [review]

**Comment by Youssef1313** [comment]
> > Yes :)
> 
> I just noticed the other callsite and have deleted the comment before your comment appeared. :)

Hmm.. Is the other callsite actually used in itself? `sourceroslyn.io` shows zero usages for all extensions in `IDiagnosticServiceExtensions`

File: src/Features/LanguageServer/Protocol/Features/Diagnostics/IDiagnosticServiceExtensions.cs [hunk 2]
```diff
@@ -5,16 +5,11 @@
...
-
         public static ValueTask<ImmutableArray<DiagnosticData>> GetPushDiagnosticsAsync(this IDiagnosticService service, DiagnosticBucket bucket, bool includeSuppressedDiagnostics, DiagnosticMode diagnosticMode, CancellationToken cancellationToken)
```
Comment (akhera99) [inline src/Features/LanguageServer/Protocol/Features/Diagnostics/IDiagnosticServiceExtensions.cs:13]: As @Youssef1313 said, this doesn't seem to have any usages either?

Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Features/Diagnostics/IDiagnosticServiceExtensions.cs:13]: i can def look into this as well.

**Pull request merged**